stuff/pipelines/test_contest_callback.py

```diff
@@ -0,0 +1,144 @@
+        window_length=0,
+        scoreboard=100,
+        points_decay_factor=0,
+        partial_score=True,
+        submissions_gap=1200,
+        penalty=0,
+        feedback='detailed',
+        penalty_type='contest_start',
+        languages='py2,py3',
+        penalty_calc_policy='sum',
+        admission_mode='private',
+        show_scoreboard_after=True,
+    )
+
+    usernames: List[str] = []
+    for number in range(5):
+        user = f'test_user_{number}'
+        client.contest.addUser(contest_alias=alias, usernameOrEmail=user)
+        usernames.append(user)
+
+    dbconn = lib.db.connect(
+        lib.db.DatabaseConnectionArguments(
+            user=credentials.MYSQL_USER,
+            password=credentials.MYSQL_PASSWORD,
+            host=credentials.MYSQL_HOST,
+            database=credentials.MYSQL_DATABASE,
+            port=credentials.MYSQL_PORT,
+            mysql_config_file=lib.db.default_config_file_path() or ''
+        )
+    )
+
+    with dbconn.cursor(buffered=True, dictionary=True) as cur:
+        cur.execute(
+            '''
+            SELECT
+                c.contest_id,
+                p.scoreboard_url
+            FROM
+                Contests c
+            INNER JOIN
+                Problemsets p
+            ON
+                p.problemset_id = c.problemset_id
+            WHERE
+                alias = %s;
+            ''', (alias,))
+        result = cur.fetchone()
+        
+    contest_id = result['contest_id']
+    scoreboard_url = result['scoreboard_url']
+    with rabbitmq_connection.connect(
+            username=credentials.OMEGAUP_USERNAME,
+            password=credentials.OMEGAUP_PASSWORD,
+            host=credentials.RABBITMQ_HOST
+    ) as channel:
+        callback = contest_callback.ContestsCallback(
+            dbconn=dbconn.conn,
+            api_token=test_constants.API_TOKEN,
+            url=test_constants.OMEGAUP_API_ENDPOINT
+        )
+        body = contest_callback.ContestCertificate(
+            contest_id=contest_id,
+            certificate_cutoff=3, # mocking a default value
+            alias=alias,
+            scoreboard_url=scoreboard_url,
+        )
+        callback(
+            _channel=channel,
+            _method=None,
+            _properties=None,
+            body=json.dumps(dataclasses.asdict(body)).encode('utf-8')
+        )
+
+    with dbconn.cursor(buffered=True, dictionary=True) as cur:
+        cur.execute(
+            '''
+            SELECT
+                i.username,
+                c.contest_place
+            FROM
+                Certificates c
+            INNER JOIN
+                Identities i
+            ON
+                i.identity_id = c.identity_id
+            INNER JOIN
+                Contests cs
+            ON
+                cs.contest_id = c.contest_id
+            WHERE
+                cs.alias = %s;
+            ''', (alias,))
+        certificates = cur.fetchall()
+
```

Comment:
```suggestion
    assert certificates
```

aquí hay que aseverar que se regresó _algo_.

Reply:
Listo